### PR TITLE
Prepare release 1.0.0

### PR DESCRIPTION
**Pull request by romanstrobl**
Merge of all changes for release `1.0.0` from `develop` to `master.

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java [hunk 1]
```diff
@@ -163,7 +162,7 @@ private void deactivatePendingActivation(Date timestamp, ActivationRecordEntity
             }
             activation.setActivationStatus(io.getlime.security.powerauth.app.server.database.model.ActivationStatus.REMOVED);
             activationHistoryServiceBehavior.saveActivationAndLogChange(activation);
-            callbackUrlBehavior.notifyCallbackListeners(activation.getApplication().getId(), activation.getActivationId());
+            callbackUrlBehavior.notifyCallbackListeners(activation.getApplication().getId(), activation);
```
Comment (petrdvorak) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java:165]: This feels... unnecessarily partial. Let's leave it and fix it in the next version (I opened an issue).

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/model/TokenInfo.java [hunk 2]
```diff
@@ -38,12 +37,10 @@ public TokenInfo() {
      * Parameterized constructor.
      * @param tokenId Token identifier.
      * @param tokenSecret Token secret.
-     * @param expires Expiration timestamp as Unix timestamp in milliseconds.
      */
-    public TokenInfo(String tokenId, String tokenSecret, Long expires) {
+    public TokenInfo(String tokenId, String tokenSecret) {
```
Comment (petrdvorak) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/model/TokenInfo.java:41]: What happened here? Is this something we should document?

**Review by petrdvorak — APPROVED** [review]
Looks OK, I have one comment that resulted in enhancement and one question regarding token expiration (did we ever even use it?).

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/model/TokenInfo.java (hunk 2 above)
Comment (romanstrobl) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/model/TokenInfo.java:41]: See: https://github.com/wultra/powerauth-server/issues/447

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/converter/v3/RecoveryPrivateKeyConverter.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_CRYPTO_PROVIDER);
+        } catch (IOException ex) {
+            logger.error(ex.getMessage(), ex);
+            // Rollback is not required, serialization is executed before database is used for writing
+            throw localizationProvider.buildExceptionForCode(ServiceError.ENCRYPTION_FAILED);
+        }
+    }
+
+    /**
+     * Derive secret key from master DB encryption key, user ID and activation ID.<br/>
```
Comment (hvge) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/converter/v3/RecoveryPrivateKeyConverter.java:215]: There's no user ID and activation ID involved in this derivation. Looks like copy-paste from similar derivation code.

**Review by hvge — APPROVED** [review]
Looks OK, I found just one javadoc discrepancy

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/converter/v3/RecoveryPrivateKeyConverter.java (hunk 3 above)
Comment (romanstrobl) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/converter/v3/RecoveryPrivateKeyConverter.java:215]: Thanks for noticing, I will fix it.

**Pull request merged**